go/interpreter/lfvm/memory_test.go

```diff
@@ -47,3 +52,232 @@ func TestExpansionCosts(t *testing.T) {
 		}
 	}
 }
+
+func TestMemory_CopyWord(t *testing.T) {
+
+	testValue := []byte{0x12, 0x34, 0x56, 0x78, 0x90, 0xab, 0xcd, 0xef}
+
+	tests := map[string]struct {
+		gas            tosca.Gas
+		offset         uint64
+		expectedTarget uint256.Int
+		expectedError  error
+	}{
+		"regular": {
+			gas:            100,
+			offset:         0,
+			expectedTarget: *uint256.NewInt(0).SetBytes8(testValue),
+			expectedError:  nil},
+		"not enough gas": {
+			gas:            6,
+			offset:         64,
+			expectedTarget: *uint256.NewInt(0),
+			expectedError:  errOutOfGas},
+	}
+
+	for name, test := range tests {
+		t.Run(name, func(t *testing.T) {
+			ctxt := getEmptyContext()
+			m := NewMemory()
+			target := uint256.NewInt(0)
+			m.store = make([]byte, 32)
+			copy(m.store[24:], testValue)
+			ctxt.gas = test.gas
+
+			err := m.CopyWord(test.offset, target, &ctxt)
+			if !errors.Is(err, test.expectedError) {
+				t.Errorf("unexpected error, want: %v, got: %v", test.expectedError, err)
+			}
+			if target.Cmp(&test.expectedTarget) != 0 {
+				t.Errorf("unexpected target value, want: %v, got: %v", test.expectedTarget, target)
+			}
+		})
+	}
+}
+
+func TestMemory_SetByte(t *testing.T) {
+
+	tests := map[string]struct {
+		value    byte
+		offset   uint64
+		gas      tosca.Gas
+		expected error
+	}{
+		"regular":        {value: 0x12, offset: 0, gas: 100, expected: nil},
+		"not enough gas": {value: 0x12, offset: 64, gas: 0, expected: errOutOfGas},
+	}
+
+	for name, test := range tests {
+		t.Run(name, func(t *testing.T) {
+			ctxt := getEmptyContext()
+			m := NewMemory()
+			ctxt.gas = test.gas
+
+			err := m.SetByte(test.offset, test.value, &ctxt)
+			if !errors.Is(err, test.expected) {
+				t.Errorf("unexpected error, want: %v, got: %v", test.expected, err)
+			}
+			if err == nil && m.store[test.offset] != test.value {
+				t.Errorf("unexpected value, want: %v, got: %v", test.value, m.store[test.offset])
+			}
+		})
+	}
+}
+
+func TestSetWordProperlyReportsNotEnoughGas(t *testing.T) {
```

Comment:
```suggestion
func TestMemory_SetWord_ReportsNotEnoughGas(t *testing.T) {
```

Reply:
This test may need to add a table like in `CopyWord`
Test in-out bounds, test that value is correct or an error.